File: src/main/java/com/elf/vipForElf/domain/contract/vo/NewContract.java
package com.elf.vipForElf.domain.contract.vo;

import com.elf.vipForElf.web.dto.NewContractDTO;
import lombok.Data;

import java.time.OffsetDateTime;

@Data
public class NewContract {
    Long realEstateListingId;
    Long realEstateAgencyId;
    String landLordId;
    OffsetDateTime moveInDt;

    public NewContract(NewContractDTO newContractDTO){
        this.realEstateAgencyId = newContractDTO.getRealEstateAgencyId();
        this.realEstateListingId = newContractDTO.getRealEstateListingId();
        this.moveInDt = newContractDTO.getMoveInDt();
        this.landLordId = newContractDTO.getLandLordId();
    }
}

File: src/main/java/com/elf/vipForElf/domain/contract/vo/NewContract.kt
package com.elf.vipForElf.domain.contract.vo

import com.elf.vipForElf.web.dto.NewContractDTO
import java.time.OffsetDateTime

data class NewContract(
    val realEstateListingId: Long?,
    val realEstateAgencyId: Long?,
    val landLordId: String?,
    val moveInDt: OffsetDateTime?
) {
    constructor(newContractDTO: NewContractDTO) : this(
        realEstateListingId = newContractDTO.realEstateListingId,
        realEstateAgencyId = newContractDTO.realEstateAgencyId,
        landLordId = newContractDTO.landLordId,
        moveInDt = newContractDTO.moveInDt
    )
}
